pkg/utils/secrets/manager/generate.go

```diff
@@ -375,19 +375,25 @@ func (m *manager) maintainLifetimeLabels(config secretutils.ConfigInterface, sec
 	}
 	desiredLabels[LabelKeyIssuedAtTime] = issuedAt
 
-	cfg, ok := config.(*secretutils.CertificateSecretConfig)
-	if !ok {
+	var dataKeyCertificate string
+	switch cfg := config.(type) {
+	case *secretutils.CertificateSecretConfig:
+		dataKeyCertificate = secretutils.DataKeyCertificate
+		if cfg.CertType == secretutils.CACert {
```

Comment:
Q -
Why the condition here is different? Previously the condition was 
```
if cfg.SigningCA == nil {
		dataKeyCertificate = secretutils.DataKeyCertificateCA
	}
```

Reply:
Because it's "more straight-forward" to check directly for the `CertType` instead of using `SigningCA == nil` as condition for a CA certificate.